Clean convex hull decomposition code

pouleyKetchoupp opened the pull request:
Following-up from https://github.com/godotengine/godot/pull/50262#issuecomment-878507055 (CC @lawnjelly)

Remove unnecessary conversion between triangle data and vertex data whenever possible.

After testing performance, it doesn't seem to make much difference because the time spent in preparing and finalizing data is negligible compared to the process in VHACD, but the code generally seems cleaner this way.

fire reviewed (commented):
Overall having lines of code deleted while doing the same thing is good.

May want someone else to review the details, but it looks good.

The code code never handled quads previously. Trying to think of why it was more complicated.

lawnjelly commented:
In general it looks great. The only suggestion I would make is, that now we are returning the raw vertices, (which in most cases is what we want), but for debugging sometimes you will want to draw these as triangles, and we have lost the connectivity information.

So I would include an optional parameter to return a triangle index list for each hull. _Or_ we could simply return a `Geometry::MeshData` for each hull, with the vertices filled out, and optionally the faces (whatever is available from VHACD function).

pouleyKetchoupp commented:
@lawnjelly I've added an extra parameter to optionally get face information in an efficient way. It's not linked to any script API, but least it's ready in case someone wants to use it for debug purpose.

reduz commented:
conflicts must be fixed

pouleyKetchoupp committed:
Clean convex hull decomposition code

Remove unnecessary conversion between triangle data and vertex data
whenever possible.
